xmlstarlet fails to build with latest libxml2

xmlstarlet is failing to build after the update to libxml2 2.14.3. A log excerpt follows:

```
2025/06/03 23:49:06 INFO config.status: executing depfiles commands
2025/06/03 23:49:06 INFO   CC       src/trans.o
2025/06/03 23:49:06 WARN src/trans.c: In function 'xsltInitLibXml':
2025/06/03 23:49:06 WARN src/trans.c:46:5: warning: 'xmlInitMemory' is deprecated [-Wdeprecated-declarations]
2025/06/03 23:49:06 WARN    46 |     xmlInitMemory();
2025/06/03 23:49:06 WARN       |     ^~~~~~~~~~~~~
2025/06/03 23:49:06 WARN In file included from src/trans.h:32,
2025/06/03 23:49:06 WARN                  from src/trans.c:4:
2025/06/03 23:49:06 WARN /usr/include/libxml2/libxml/xmlmemory.h:133:9: note: declared here
2025/06/03 23:49:06 WARN   133 |         xmlInitMemory   (void);
2025/06/03 23:49:06 WARN       |         ^~~~~~~~~~~~~
2025/06/03 23:49:06 WARN src/trans.c:53:5: warning: 'xmlLineNumbersDefault' is deprecated [-Wdeprecated-declarations]
2025/06/03 23:49:06 WARN    53 |     xmlLineNumbersDefault(1);
2025/06/03 23:49:06 WARN       |     ^~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:06 WARN In file included from /usr/include/libxml2/libxml/tree.h:17,
2025/06/03 23:49:06 WARN                  from /usr/include/libxml2/libxml/debugXML.h:15,
2025/06/03 23:49:06 WARN                  from src/trans.h:33:
2025/06/03 23:49:06 WARN /usr/include/libxml2/libxml/parser.h:1125:17: note: declared here
2025/06/03 23:49:06 WARN  1125 |                 xmlLineNumbersDefault   (int val);
2025/06/03 23:49:06 WARN       |                 ^~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:06 WARN src/trans.c:73:5: warning: 'xmlPedanticParserDefault' is deprecated [-Wdeprecated-declarations]
2025/06/03 23:49:06 WARN    73 |     xmlPedanticParserDefault(0);
2025/06/03 23:49:06 WARN       |     ^~~~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:06 WARN /usr/include/libxml2/libxml/parser.h:1119:17: note: declared here
2025/06/03 23:49:06 WARN  1119 |                 xmlPedanticParserDefault(int val);
2025/06/03 23:49:06 WARN       |                 ^~~~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:06 WARN src/trans.c:75:5: warning: '__xmlGetWarningsDefaultValue' is deprecated [-Wdeprecated-declarations]
2025/06/03 23:49:06 WARN    75 |     xmlGetWarningsDefaultValue = 1;
2025/06/03 23:49:06 WARN       |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:06 WARN /usr/include/libxml2/libxml/parser.h:1028:16: note: declared here
2025/06/03 23:49:06 WARN  1028 | XMLPUBFUN int *__xmlGetWarningsDefaultValue(void);
2025/06/03 23:49:06 WARN       |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:06 WARN src/trans.c:77:5: warning: '__xmlLoadExtDtdDefaultValue' is deprecated [-Wdeprecated-declarations]
2025/06/03 23:49:06 WARN    77 |     xmlLoadExtDtdDefaultValue = 1;
2025/06/03 23:49:06 WARN       |     ^~~~~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:06 WARN /usr/include/libxml2/libxml/parser.h:1034:16: note: declared here
2025/06/03 23:49:06 WARN  1034 | XMLPUBFUN int *__xmlLoadExtDtdDefaultValue(void);
2025/06/03 23:49:06 WARN       |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:06 WARN src/trans.c:84:9: warning: '__xmlLoadExtDtdDefaultValue' is deprecated [-Wdeprecated-declarations]
2025/06/03 23:49:06 WARN    84 |         xmlLoadExtDtdDefaultValue = XML_DETECT_IDS | XML_COMPLETE_ATTRS;
2025/06/03 23:49:06 WARN       |         ^~~~~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:06 WARN /usr/include/libxml2/libxml/parser.h:1034:16: note: declared here
2025/06/03 23:49:06 WARN  1034 | XMLPUBFUN int *__xmlLoadExtDtdDefaultValue(void);
2025/06/03 23:49:06 WARN       |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:06 WARN src/trans.c:88:9: warning: '__xmlLoadExtDtdDefaultValue' is deprecated [-Wdeprecated-declarations]
2025/06/03 23:49:06 WARN    88 |         xmlLoadExtDtdDefaultValue = 0;
2025/06/03 23:49:06 WARN       |         ^~~~~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:06 WARN /usr/include/libxml2/libxml/parser.h:1034:16: note: declared here
2025/06/03 23:49:06 WARN  1034 | XMLPUBFUN int *__xmlLoadExtDtdDefaultValue(void);
2025/06/03 23:49:06 WARN       |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:07 INFO version.h: #define VERSION "1.6.1-dirty"
2025/06/03 23:49:07 INFO   CC       src/xml.o
2025/06/03 23:49:07 WARN src/xml.c: In function 'main':
2025/06/03 23:49:07 WARN src/xml.c:300:43: warning: passing argument 2 of 'xmlSetStructuredErrorFunc' from incompatible pointer type [-Wincompatible-pointer-types]
2025/06/03 23:49:07 WARN   300 |     xmlSetStructuredErrorFunc(&errorInfo, reportError);
2025/06/03 23:49:07 WARN       |                                           ^~~~~~~~~~~
2025/06/03 23:49:07 WARN       |                                           |
2025/06/03 23:49:07 WARN       |                                           void (*)(void *, xmlError *) {aka void (*)(void *, struct _xmlError *)}
2025/06/03 23:49:07 WARN In file included from /usr/include/libxml2/libxml/valid.h:16,
2025/06/03 23:49:07 WARN                  from /usr/include/libxml2/libxml/parser.h:20,
2025/06/03 23:49:07 WARN                  from /usr/include/libxml2/libxml/tree.h:17,
2025/06/03 23:49:07 WARN                  from /usr/include/libxslt/xslt.h:13,
2025/06/03 23:49:07 WARN                  from src/xml.c:37:
2025/06/03 23:49:07 WARN /usr/include/libxml2/libxml/xmlerror.h:904:57: note: expected 'xmlStructuredErrorFunc' {aka 'void (*)(void *, const struct _xmlError *)'} but argument is of type 'void (*)(void *, xmlError *)' {aka 'void (*)(void *, struct _xmlError *)'}
2025/06/03 23:49:07 WARN   904 |                                  xmlStructuredErrorFunc handler);
2025/06/03 23:49:07 WARN       |                                  ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
2025/06/03 23:49:07 WARN src/xml.c:107:6: note: 'reportError' declared here
2025/06/03 23:49:07 WARN   107 | void reportError(void *ptr, xmlErrorPtr error)
2025/06/03 23:49:07 WARN       |      ^~~~~~~~~~~
2025/06/03 23:49:07 WARN /usr/include/libxml2/libxml/xmlerror.h:870:16: note: 'xmlStructuredErrorFunc' declared here
2025/06/03 23:49:07 WARN   870 | typedef void (*xmlStructuredErrorFunc) (void *userData, const xmlError *error);
2025/06/03 23:49:07 WARN       |                ^~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:07 INFO   CC       src/xml_C14N.o
2025/06/03 23:49:07 WARN src/xml_C14N.c: In function 'c14nMain':
2025/06/03 23:49:07 WARN src/xml_C14N.c:171:5: warning: 'xmlMemoryDump' is deprecated [-Wdeprecated-declarations]
2025/06/03 23:49:07 WARN   171 |     xmlMemoryDump();
2025/06/03 23:49:07 WARN       |     ^~~~~~~~~~~~~
2025/06/03 23:49:07 WARN In file included from src/xml_C14N.c:23:
2025/06/03 23:49:07 WARN /usr/include/libxml2/libxml/xmlmemory.h:161:9: note: declared here
2025/06/03 23:49:07 WARN   161 |         xmlMemoryDump   (void);
2025/06/03 23:49:07 WARN       |         ^~~~~~~~~~~~~
2025/06/03 23:49:07 WARN src/xml_C14N.c: In function 'load_xpath_expr':
2025/06/03 23:49:07 WARN src/xml_C14N.c:240:5: warning: '__xmlLoadExtDtdDefaultValue' is deprecated [-Wdeprecated-declarations]
2025/06/03 23:49:07 WARN   240 |     xmlLoadExtDtdDefaultValue = XML_DETECT_IDS | XML_COMPLETE_ATTRS;
2025/06/03 23:49:07 WARN       |     ^~~~~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:07 WARN In file included from src/xml_C14N.c:24:
2025/06/03 23:49:07 WARN /usr/include/libxml2/libxml/parser.h:1034:16: note: declared here
2025/06/03 23:49:07 WARN  1034 | XMLPUBFUN int *__xmlLoadExtDtdDefaultValue(void);
2025/06/03 23:49:07 WARN       |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:07 WARN src/xml_C14N.c:241:5: warning: 'xmlSubstituteEntitiesDefault' is deprecated [-Wdeprecated-declarations]
2025/06/03 23:49:07 WARN   241 |     xmlSubstituteEntitiesDefault(1);
2025/06/03 23:49:07 WARN       |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:07 WARN /usr/include/libxml2/libxml/parser.h:1106:17: note: declared here
2025/06/03 23:49:07 WARN  1106 |                 xmlSubstituteEntitiesDefault(int val);
2025/06/03 23:49:07 WARN       |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:07 INFO   CC       src/xml_depyx.o
2025/06/03 23:49:07 INFO   CC       src/xml_edit.o
2025/06/03 23:49:07 WARN src/xml_edit.c: In function 'edProcess':
2025/06/03 23:49:07 WARN src/xml_edit.c:458:5: warning: 'xmlDeregisterNodeDefault' is deprecated [-Wdeprecated-declarations]
2025/06/03 23:49:07 WARN   458 |     xmlDeregisterNodeDefault(&removeNodeFromPrev);
2025/06/03 23:49:07 WARN       |     ^~~~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:07 WARN In file included from /usr/include/libxml2/libxml/parser.h:16,
2025/06/03 23:49:07 WARN                  from /usr/include/libxml2/libxml/tree.h:17,
2025/06/03 23:49:07 WARN                  from /usr/include/libxml2/libxml/debugXML.h:15,
2025/06/03 23:49:07 WARN                  from src/xml_edit.c:36:
2025/06/03 23:49:07 WARN /usr/include/libxml2/libxml/tree.h:1311:13: note: declared here
2025/06/03 23:49:07 WARN  1311 |             xmlDeregisterNodeDefault    (xmlDeregisterNodeFunc func);
2025/06/03 23:49:07 WARN       |             ^~~~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:07 WARN src/xml_edit.c:533:5: warning: 'xmlDeregisterNodeDefault' is deprecated [-Wdeprecated-declarations]
2025/06/03 23:49:07 WARN   533 |     xmlDeregisterNodeDefault(NULL);
2025/06/03 23:49:07 WARN       |     ^~~~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:07 WARN /usr/include/libxml2/libxml/tree.h:1311:13: note: declared here
2025/06/03 23:49:07 WARN  1311 |             xmlDeregisterNodeDefault    (xmlDeregisterNodeFunc func);
2025/06/03 23:49:07 WARN       |             ^~~~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:07 WARN src/xml_edit.c: In function 'edOutput':
2025/06/03 23:49:07 WARN src/xml_edit.c:561:9: warning: 'xmlCleanupGlobals' is deprecated [-Wdeprecated-declarations]
2025/06/03 23:49:07 WARN   561 |         xmlCleanupGlobals();
2025/06/03 23:49:07 WARN       |         ^~~~~~~~~~~~~~~~~
2025/06/03 23:49:07 WARN /usr/include/libxml2/libxml/parser.h:1079:17: note: declared here
2025/06/03 23:49:07 WARN  1079 |                 xmlCleanupGlobals       (void);
2025/06/03 23:49:07 WARN       |                 ^~~~~~~~~~~~~~~~~
2025/06/03 23:49:07 WARN src/xml_edit.c: In function 'edMain':
2025/06/03 23:49:07 WARN src/xml_edit.c:755:5: warning: 'xmlCleanupGlobals' is deprecated [-Wdeprecated-declarations]
2025/06/03 23:49:07 WARN   755 |     xmlCleanupGlobals();
2025/06/03 23:49:07 WARN src/xml_format.c: In function 'foInitLibXml':
2025/06/03 23:49:07 WARN src/xml_format.c:114:5: warning: 'xmlInitMemory' is deprecated [-Wdeprecated-declarations]
2025/06/03 23:49:07 WARN   114 |     xmlInitMemory();
2025/06/03 23:49:07 WARN       |     ^~~~~~~~~~~~~
2025/06/03 23:49:07 WARN In file included from src/xml_format.c:35:
2025/06/03 23:49:07 WARN /usr/include/libxml2/libxml/xmlmemory.h:133:9: note: declared here
2025/06/03 23:49:07 WARN   133 |         xmlInitMemory   (void);
2025/06/03 23:49:07 WARN       |         ^~~~~~~~~~~~~
2025/06/03 23:49:07 WARN src/xml_format.c:121:5: warning: 'xmlLineNumbersDefault' is deprecated [-Wdeprecated-declarations]
2025/06/03 23:49:07 WARN   121 |     xmlLineNumbersDefault(1);
2025/06/03 23:49:07 WARN       |     ^~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:07 WARN In file included from /usr/include/libxml2/libxml/tree.h:17,
2025/06/03 23:49:07 WARN                  from /usr/include/libxml2/libxml/debugXML.h:15,
2025/06/03 23:49:07 WARN                  from src/xml_format.c:36:
2025/06/03 23:49:07 WARN /usr/include/libxml2/libxml/parser.h:1125:17: note: declared here
2025/06/03 23:49:07 WARN  1125 |                 xmlLineNumbersDefault   (int val);
2025/06/03 23:49:07 WARN       |                 ^~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:07 WARN src/xml_format.c:123:5: warning: 'xmlSubstituteEntitiesDefault' is deprecated [-Wdeprecated-declarations]
2025/06/03 23:49:07 WARN   123 |     xmlSubstituteEntitiesDefault(1);
2025/06/03 23:49:07 WARN       |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:07 WARN /usr/include/libxml2/libxml/parser.h:1106:17: note: declared here
2025/06/03 23:49:07 WARN  1106 |                 xmlSubstituteEntitiesDefault(int val);
2025/06/03 23:49:07 WARN       |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:07 WARN src/xml_format.c:125:5: warning: 'xmlPedanticParserDefault' is deprecated [-Wdeprecated-declarations]
2025/06/03 23:49:07 WARN   125 |     xmlPedanticParserDefault(0);
2025/06/03 23:49:07 WARN       |     ^~~~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:07 WARN /usr/include/libxml2/libxml/parser.h:1119:17: note: declared here
2025/06/03 23:49:07 WARN  1119 |                 xmlPedanticParserDefault(int val);
2025/06/03 23:49:07 WARN       |                 ^~~~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:07 WARN src/xml_format.c:127:5: warning: '__xmlGetWarningsDefaultValue' is deprecated [-Wdeprecated-declarations]
2025/06/03 23:49:07 WARN   127 |     xmlGetWarningsDefaultValue = 1;
2025/06/03 23:49:07 WARN       |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:07 WARN /usr/include/libxml2/libxml/parser.h:1028:16: note: declared here
2025/06/03 23:49:07 WARN  1028 | XMLPUBFUN int *__xmlGetWarningsDefaultValue(void);
2025/06/03 23:49:07 WARN       |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:07 WARN src/xml_format.c:128:5: warning: '__xmlDoValidityCheckingDefaultValue' is deprecated [-Wdeprecated-declarations]
2025/06/03 23:49:07 WARN   128 |     xmlDoValidityCheckingDefaultValue = 0;
2025/06/03 23:49:07 WARN       |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:07 WARN /usr/include/libxml2/libxml/parser.h:1026:16: note: declared here
2025/06/03 23:49:07 WARN  1026 | XMLPUBFUN int *__xmlDoValidityCheckingDefaultValue(void);
2025/06/03 23:49:07 WARN       |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:07 WARN src/xml_format.c:129:5: warning: '__xmlLoadExtDtdDefaultValue' is deprecated [-Wdeprecated-declarations]
2025/06/03 23:49:07 WARN   129 |     xmlLoadExtDtdDefaultValue = 0;
2025/06/03 23:49:07 WARN       |     ^~~~~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:07 WARN /usr/include/libxml2/libxml/parser.h:1034:16: note: declared here
2025/06/03 23:49:07 WARN  1034 | XMLPUBFUN int *__xmlLoadExtDtdDefaultValue(void);
2025/06/03 23:49:07 WARN       |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:07 INFO   CC       src/xml_ls.o
2025/06/03 23:49:07 WARN src/xml_ls.c: In function 'xml_print_dir':
2025/06/03 23:49:07 WARN src/xml_ls.c:149:22: warning: the comparison will always evaluate as 'false' for the address of 'd_name' will never be NULL [-Waddress]
2025/06/03 23:49:07 WARN   149 |       if ((d->d_name == NULL) || !strcmp(d->d_name, ".") || !strcmp(d->d_name, ".."))
2025/06/03 23:49:07 WARN       |                      ^~
2025/06/03 23:49:07 WARN In file included from /usr/include/dirent.h:61,
2025/06/03 23:49:07 WARN                  from src/xml_ls.c:33:
2025/06/03 23:49:07 WARN /usr/include/bits/dirent.h:33:10: note: 'd_name' declared here
2025/06/03 23:49:07 WARN    33 |     char d_name[256];           /* We must not include limits.h! */
2025/06/03 23:49:07 WARN       |          ^~~~~~
2025/06/03 23:49:07 WARN src/xml_ls.c:140:4: warning: ignoring return value of 'chdir' declared with attribute 'warn_unused_result' [-Wunused-result]
2025/06/03 23:49:07 WARN   140 |    chdir(dir);
2025/06/03 23:49:07 WARN       |    ^~~~~~~~~~
2025/06/03 23:49:07 INFO   CC       src/xml_pyx.o
2025/06/03 23:49:07 WARN       |     ^~~~~~~~~~~~~~~~~
2025/06/03 23:49:07 WARN /usr/include/libxml2/libxml/parser.h:1079:17: note: declared here
2025/06/03 23:49:07 WARN  1079 |                 xmlCleanupGlobals       (void);
2025/06/03 23:49:07 WARN       |                 ^~~~~~~~~~~~~~~~~
2025/06/03 23:49:07 WARN src/xml_edit.c: In function 'edInsert':
2025/06/03 23:49:07 WARN src/xml_edit.c:371:9: warning: 'node' may be used uninitialized [-Wmaybe-uninitialized]
2025/06/03 23:49:07 WARN   371 |         xmlXPathNodeSetAdd(previous_insertion, node);
2025/06/03 23:49:07 WARN       |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:07 WARN src/xml_edit.c:339:20: note: 'node' was declared here
2025/06/03 23:49:07 WARN   339 |         xmlNodePtr node;
2025/06/03 23:49:07 WARN       |                    ^~~~
2025/06/03 23:49:07 INFO   CC       src/xml_elem.o
2025/06/03 23:49:07 WARN src/xml_elem.c: In function 'elMain':
2025/06/03 23:49:07 WARN src/xml_elem.c:271:27: warning: passing argument 2 of 'xmlHashScan' from incompatible pointer type [-Wincompatible-pointer-types]
2025/06/03 23:49:07 WARN   271 |         xmlHashScan(uniq, hash_key_put, &lines);
2025/06/03 23:49:07 WARN       |                           ^~~~~~~~~~~~
2025/06/03 23:49:07 WARN       |                           |
2025/06/03 23:49:07 WARN       |                           void (*)(void *, void *, xmlChar *) {aka void (*)(void *, void *, unsigned char *)}
2025/06/03 23:49:07 WARN In file included from src/xml_elem.c:32:
2025/06/03 23:49:07 WARN /usr/include/libxml2/libxml/hash.h:228:57: note: expected 'xmlHashScanner' {aka 'void (*)(void *, void *, const unsigned char *)'} but argument is of type 'void (*)(void *, void *, xmlChar *)' {aka 'void (*)(void *, void *, unsigned char *)'}
2025/06/03 23:49:07 WARN   228 |                                          xmlHashScanner scan,
2025/06/03 23:49:07 WARN       |                                          ~~~~~~~~~~~~~~~^~~~
2025/06/03 23:49:07 WARN src/xml_elem.c:189:1: note: 'hash_key_put' declared here
2025/06/03 23:49:07 WARN   189 | hash_key_put(void *payload, void *data, xmlChar *name)
2025/06/03 23:49:07 WARN       | ^~~~~~~~~~~~
2025/06/03 23:49:07 WARN /usr/include/libxml2/libxml/hash.h:79:16: note: 'xmlHashScanner' declared here
2025/06/03 23:49:07 WARN    79 | typedef void (*xmlHashScanner)(void *payload, void *data, const xmlChar *name);
2025/06/03 23:49:07 WARN       |                ^~~~~~~~~~~~~~
2025/06/03 23:49:07 INFO   CC       src/xml_escape.o
2025/06/03 23:49:07 INFO   CC       src/xml_format.o

2025/06/03 23:49:08 WARN src/xml_pyx.c:203:36: error: expected ';', ',' or ')' before 'ATTRIBUTE_UNUSED'
2025/06/03 23:49:08 WARN   203 | pyxExternalSubsetHandler(void *ctx ATTRIBUTE_UNUSED, const xmlChar *name,
2025/06/03 23:49:08 WARN       |                                    ^~~~~~~~~~~~~~~~
2025/06/03 23:49:08 WARN src/xml_pyx.c:218:29: error: expected ';', ',' or ')' before 'ATTRIBUTE_UNUSED'
2025/06/03 23:49:08 WARN   218 | pyxCommentHandler(void *ctx ATTRIBUTE_UNUSED, const xmlChar *value)
2025/06/03 23:49:08 WARN       |                             ^~~~~~~~~~~~~~~~
2025/06/03 23:49:08 WARN src/xml_pyx.c:226:32: error: expected ';', ',' or ')' before 'ATTRIBUTE_UNUSED'
2025/06/03 23:49:08 WARN   226 | pyxCdataBlockHandler(void *ctx ATTRIBUTE_UNUSED, const xmlChar *value, int len)
2025/06/03 23:49:08 WARN       |                                ^~~~~~~~~~~~~~~~
2025/06/03 23:49:08 WARN src/xml_pyx.c: In function 'pyxMain':
2025/06/03 23:49:08 WARN src/xml_pyx.c:291:29: error: 'pyxExternalSubsetHandler' undeclared (first use in this function)
2025/06/03 23:49:08 WARN   291 |     pyxSAX.externalSubset = pyxExternalSubsetHandler;
2025/06/03 23:49:08 WARN       |                             ^~~~~~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:08 WARN src/xml_pyx.c:291:29: note: each undeclared identifier is reported only once for each function it appears in
2025/06/03 23:49:08 WARN src/xml_pyx.c:292:22: error: 'pyxCommentHandler' undeclared (first use in this function)
2025/06/03 23:49:08 WARN   292 |     pyxSAX.comment = pyxCommentHandler;
2025/06/03 23:49:08 WARN       |                      ^~~~~~~~~~~~~~~~~
2025/06/03 23:49:08 WARN src/xml_pyx.c:293:25: error: 'pyxCdataBlockHandler' undeclared (first use in this function)
2025/06/03 23:49:08 WARN   293 |     pyxSAX.cdataBlock = pyxCdataBlockHandler;
2025/06/03 23:49:08 WARN       |                         ^~~~~~~~~~~~~~~~~~~~
2025/06/03 23:49:08 WARN make: *** [Makefile:947: src/xml_pyx.o] Error 1
2025/06/03 23:49:08 ERRO failed to build package: unable to run package xmlstarlet pipeline: unable to run pipeline: exit status 2
make[1]: *** [Makefile:157: packages/x86_64/xmlstarlet-1.6.1-r5.apk] Error 1
make[1]: Leaving directory '/home/user/src/wolfi-os'
make: *** [Makefile:151: package/xmlstarlet] Error 2
```

--- a/xmlstarlet.yaml
+++ b/xmlstarlet.yaml
@@ -1,7 +1,7 @@
 package:
   name: xmlstarlet
   version: 1.6.1
-  epoch: 5
+  epoch: 6
   description: A set of tools to transform, query, validate, and edit XML documents
   copyright:
     - license: MIT
@@ -29,6 +29,14 @@ pipeline:
   - uses: patch
     with:
       patches: docbook-update.patch
+
+  - uses: patch
+    with:
+      patches: 0001-Fix-build-with-libxml2-2.12.patch
+
+  - uses: patch
+    with:
+      patches: no-attribute-unused.diff
 
   - runs: |
       autoreconf -sif

--- a/xmlstarlet/0001-Fix-build-with-libxml2-2.12.patch
+++ b/xmlstarlet/0001-Fix-build-with-libxml2-2.12.patch
@@ -1,0 +1,77 @@
+Origin: https://sourceforge.net/p/xmlstar/patches/_discuss/thread/890e29655a/66ca/attachment/0001-Fix-build-with-libxml2-2.12.patch
+
+diff --git a/src/trans.h b/src/trans.h
+index 016304e..395462c 100644
+--- a/src/trans.h
++++ b/src/trans.h
+@@ -36,6 +36,7 @@ THE SOFTWARE.
+ #include <libxml/xinclude.h>
+ #include <libxml/parserInternals.h>
+ #include <libxml/uri.h>
++#include <libxml/globals.h>
+ 
+ #include <libxslt/xslt.h>
+ #include <libxslt/xsltInternals.h>
+diff --git a/src/xml.c b/src/xml.c
+index cf47cc2..e6f19c1 100644
+--- a/src/xml.c
++++ b/src/xml.c
+@@ -104,7 +104,7 @@ void reportGenericError(void* ctx, const char * msg, ...) {
+ /* by default all errors are reported */
+ static ErrorInfo errorInfo = { NULL, NULL, VERBOSE, CONTINUE };
+ 
+-void reportError(void *ptr, xmlErrorPtr error)
++void reportError(void *ptr, const xmlError *error)
+ {
+     ErrorInfo *errorInfo = (ErrorInfo*) ptr;
+     assert(errorInfo);
+diff --git a/src/xml_edit.c b/src/xml_edit.c
+index e199178..b3da8a4 100644
+--- a/src/xml_edit.c
++++ b/src/xml_edit.c
+@@ -42,6 +42,7 @@ THE SOFTWARE.
+ #include <libxml/xpointer.h>
+ #include <libxml/parserInternals.h>
+ #include <libxml/uri.h>
++#include <libxml/globals.h>
+ #include <libexslt/exslt.h>
+ 
+ #include "xmlstar.h"
+diff --git a/src/xml_elem.c b/src/xml_elem.c
+index 024e62a..a73038a 100644
+--- a/src/xml_elem.c
++++ b/src/xml_elem.c
+@@ -186,7 +186,7 @@ typedef struct {
+  * put @name into @data->array[@data->offset]
+  */
+ static void
+-hash_key_put(void *payload, void *data, xmlChar *name)
++hash_key_put(void *payload, void *data, const xmlChar *name)
+ {
+     ArrayDest *dest = data;
+     dest->array[dest->offset++] = name;
+diff --git a/src/xml_format.c b/src/xml_format.c
+index e9da4df..ddd6614 100644
+--- a/src/xml_format.c
++++ b/src/xml_format.c
+@@ -42,6 +42,7 @@ THE SOFTWARE.
+ #include <libxml/xpointer.h>
+ #include <libxml/parserInternals.h>
+ #include <libxml/uri.h>
++#include <libxml/globals.h>
+ 
+ #include "xmlstar.h"
+ 
+diff --git a/src/xmlstar.h b/src/xmlstar.h
+index 3e1eed3..e8d7177 100644
+--- a/src/xmlstar.h
++++ b/src/xmlstar.h
+@@ -32,7 +32,7 @@ typedef struct _errorInfo {
+     ErrorStop stop;
+ } ErrorInfo;
+ 
+-void reportError(void *ptr, xmlErrorPtr error);
++void reportError(void *ptr, const xmlError *error);
+ void suppressErrors(void);
+ 
+ typedef struct _gOptions {

--- a/xmlstarlet/no-attribute-unused.diff
+++ b/xmlstarlet/no-attribute-unused.diff
@@ -1,0 +1,31 @@
+Origin: Ubuntu 1.6.1-4ubuntu1
+
+--- a/src/xml_pyx.c
++++ b/src/xml_pyx.c
+@@ -200,7 +200,7 @@ pyxExternalEntityReferenceHandler(void*
+ }
+ 
+ static void
+-pyxExternalSubsetHandler(void *ctx ATTRIBUTE_UNUSED, const xmlChar *name,
++pyxExternalSubsetHandler(void *ctx, const xmlChar *name,
+                          const xmlChar *ExternalID, const xmlChar *SystemID)
+ {
+     fprintf(stdout, "D %s PUBLIC", name); /* TODO: re-check */
+@@ -215,7 +215,7 @@ pyxExternalSubsetHandler(void *ctx ATTRI
+ }
+ 
+ static void
+-pyxCommentHandler(void *ctx ATTRIBUTE_UNUSED, const xmlChar *value)
++pyxCommentHandler(void *ctx, const xmlChar *value)
+ {
+     fprintf(stdout, "%s", "C");
+     SanitizeData(value, xmlStrlen(value));
+@@ -223,7 +223,7 @@ pyxCommentHandler(void *ctx ATTRIBUTE_UN
+ }
+ 
+ static void
+-pyxCdataBlockHandler(void *ctx ATTRIBUTE_UNUSED, const xmlChar *value, int len)
++pyxCdataBlockHandler(void *ctx, const xmlChar *value, int len)
+ {
+     fprintf(stdout, "%s", "[");
+     SanitizeData(value, len);

